cli/azd/pkg/infra/scope_test.go

```diff
@@ -19,37 +23,64 @@ func TestScopeGetDeployment(t *testing.T) {
 		Value: "https://www.myapp.com",
 	}
 
-	deployment := azcli.AzCliDeploymentResult{
-		Properties: azcli.AzCliDeploymentResultProperties{
-			Outputs: outputs,
+	// mocked response for get deployment from subscription
+	deploymentWithOptions := &armresources.DeploymentsClientGetAtSubscriptionScopeResponse{
+		DeploymentExtended: armresources.DeploymentExtended{
+			Properties: &armresources.DeploymentPropertiesExtended{
+				Outputs: outputs,
+			},
+		},
+	}
+	deploymentResourceGroupWithOptions := &armresources.DeploymentsClientGetResponse{
+		DeploymentExtended: armresources.DeploymentExtended{
+			Properties: &armresources.DeploymentPropertiesExtended{
+				Outputs: outputs,
+			},
 		},
 	}
-	deploymentBytes, _ := json.Marshal(deployment)
 
 	t.Run("SubscriptionScopeSuccess", func(t *testing.T) {
 		mockContext := mocks.NewMockContext(context.Background())
-		mockContext.CommandRunner.When(func(args exec.RunArgs, command string) bool {
-			return strings.Contains(command, "az deployment sub show")
-		}).Respond(exec.NewRunResult(0, string(deploymentBytes), ""))
+		mockContext.HttpClient.When(func(request *http.Request) bool {
+			return request.Method == http.MethodGet && strings.Contains(request.URL.Path, "/subscriptions")
+		}).RespondFn(func(request *http.Request) (*http.Response, error) {
+			subscriptionsListBytes, _ := json.Marshal(deploymentWithOptions)
+
+			return &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       io.NopCloser(bytes.NewBuffer(subscriptionsListBytes)),
+			}, nil
+		})
 
 		scope := NewSubscriptionScope(*mockContext.Context, "eastus2", "SUBSCRIPTION_ID", "DEPLOYMENT_NAME")
 
 		deployment, err := scope.GetDeployment(*mockContext.Context)
 		require.NoError(t, err)
-		require.Equal(t, outputs["APP_URL"].Value, deployment.Properties.Outputs["APP_URL"].Value)
+		responseOutputs := deployment.Properties.Outputs.(map[string]interface{})["APP_URL"].(map[string]interface{})
+		require.Equal(t, outputs["APP_URL"].Value, responseOutputs["value"].(string))
+		require.Equal(t, outputs["APP_URL"].Type, responseOutputs["type"].(string))
 	})
 
 	t.Run("ResourceGroupScopeSuccess", func(t *testing.T) {
 		mockContext := mocks.NewMockContext(context.Background())
-		mockContext.CommandRunner.When(func(args exec.RunArgs, command string) bool {
-			return strings.Contains(command, "az deployment group show")
-		}).Respond(exec.NewRunResult(0, string(deploymentBytes), ""))
+		mockContext.HttpClient.When(func(request *http.Request) bool {
+			return request.Method == http.MethodGet && strings.Contains(request.URL.Path, "/subscriptions")
```

Comment:
For the resource group scope I would have expected to see the http client path contain both a resource groups reference in addition to subscription??

Reply:
I can make it more explicit, sure